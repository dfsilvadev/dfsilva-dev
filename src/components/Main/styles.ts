'use client';

import styled from 'styled-components';

export const MainContent = styled.main`
  background-color: #f2f3f5;
  width: 100vw;
  height: 100vh;
  padding: 3rem;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
