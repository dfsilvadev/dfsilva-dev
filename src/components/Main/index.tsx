import Image from 'next/image';

import * as S from './styles';

const Main = () => (
  <S.MainContent>
    <h1>Boilerplate</h1>
    <div>
      <Image
        src="/next.svg"
        alt="Next.js Logo"
        width={180}
        height={37}
        priority
      />
    </div>
  </S.MainContent>
);

export default Main;
