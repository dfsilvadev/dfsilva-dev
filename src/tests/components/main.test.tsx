import { render, screen } from '@testing-library/react';

import Main from '@/components/Main';

describe('<Main />', () => {
  it('should render the heading and next logo', () => {
    const { container } = render(<Main />);

    const heading = screen.getByRole('heading', { name: /boilerplate/i });
    const nextLogo = screen.getByRole('img', { name: /next.js logo/i });

    expect(heading).toBeInTheDocument();

    expect(nextLogo).toBeInTheDocument();
    expect(nextLogo).toHaveAttribute('src', '/next.svg');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the background color currectly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#f2f3f5' });
  });
});
