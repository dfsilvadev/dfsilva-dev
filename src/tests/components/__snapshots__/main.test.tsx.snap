// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading and next logo 1`] = `
.c0 {
  background-color: #f2f3f5;
  width: 100vw;
  height: 100vh;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

<main
  class="c0"
>
  <h1>
    Boilerplate
  </h1>
  <div>
    <img
      alt="Next.js Logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="37"
      src="/next.svg"
      style="color: transparent;"
      width="180"
    />
  </div>
</main>
`;
